### DISCUSSION TITLE
Adding new speaker

### DISCUSSION THREAD

**Discussion by clonedSemicolon** [coqui-ai/TTS#1805] [General Q&A]
Hey, the model works fine for the selected speakers list. I was just wondering if can i add my own voice as speaker or any custom voice in the speaker list?

**Comment by erogol** [comment]
You can add a new speaker but the current API does not support it. You need to manually add a new row to the speaker embedding layer. If you are using d-vectors then you don't need to change anything. Just compute d-vectors for your new speaker and train the model.

**Reply by Onkarsus13** [reply]
I tried this approach but when I add a new speaker id and finetune the model, the model is forgetting the previous speaker accents, audio of older speaker is becoming streched

**Reply by kin0303** [reply]
> I tried this approach but when I add a new speaker id and finetune the model, the model is forgetting the previous speaker accents, audio of older speaker is becoming streched

Hi, how you do that? I'm a newbie

**Reply by Onkarsus13** [reply]
Hii you just add new speaker and new speaker.pth and train it

**Reply by kin0303** [reply]
> Hii you just add new speaker and new speaker.pth and train it

How to do that?

**Reply by kin0303** [reply]
> speaker.pth

How to make file speaker.pth?

**Reply by Onkarsus13** [reply]
Itterate over the speaker list
make the dict like this and finally append the new speaker at last
{
'speaker1':0,
'speaker2':1
.
.
.
'speakerN':N-1
}

**Reply by kin0303** [reply]
> Itterate over the speaker list make the dict like this and finally append the new speaker at last { 'speaker1':0, 'speaker2':1 . . . 'speakerN':N-1 }

I want to train using a single speaker, and when the output sounds good, I want to add a new speaker. Can I do that? And the model will be multi-speaker model using this way. Or I just can do it when I'm training multi-speaker?

**Reply by kin0303** [reply]
> Itterate over the speaker list
> make the dict like this and finally append the new speaker at last

My speaker.pth look like this
![Screenshot from 2022-10-24 11-29-13](https://user-images.githubusercontent.com/103550046/197448805-4a90fc80-24b7-42a5-b633-e16927639021.png)

**Reply by kin0303** [reply]
> > Itterate over the speaker list
> > make the dict like this and finally append the new speaker at last
> 
> My speaker.pth look like this ![Screenshot from 2022-10-24 11-29-13](https://user-images.githubusercontent.com/103550046/197448805-4a90fc80-24b7-42a5-b633-e16927639021.png)

File data.pkl look like this

```
8002 7d71 0028 5805 0000 0041 6469 746f
7101 4b00 5804 0000 0042 6162 7971 024b
0158 0400 0000 4265 6c61 7103 4b02 5805
0000 0048 616c 7761 7104 4b03 752e 
```
and when running using pickle, the file look like this:
{'Anna': 0, 'Beny': 1, 'Elsa': 2, 'Olaf: 3}

So, I need to compute d-vectors right? And adding to data.pkl? Is it computing d-vector using this script ```python TTS/bin/compute_embeddings.py```?
